lib/ecto/query.ex

```diff
@@ -95,6 +95,13 @@ defmodule Ecto.Query do
   For this reason, we will use schemas on the remaining examples but
   remember Ecto does not require them in order to write queries.
 
+  ## `nil` comparison
+
+  Nil comparison with a infix operators is forbidden as it is unsafe
+  and would expose security risks. To check that value is `nil` use `is_nil/1`.
+
+  If you want to dynamically extend your query, use a query composition.
```

Comment:
I don't think this is really related and composition is the next section anyway :)

Reply:
@ericmj 
That was my plan, to make docs look like a continuous story :)